src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -375,6 +375,13 @@ public void removeObsoleteServices() throws InterruptedException, ExecutionExcep
                 serviceClosedFuture.get();
             }
             servicesToRemove.forEach(serviceName -> {
+                try {
+                    GreengrassService eg = kernel.locate(serviceName);
+                    kernel.getContext().remove(eg.getClass());
+                } catch (ServiceLoadException e) {
+                    logger.atError(MERGE_ERROR_LOG_EVENT_KEY).setCause(e).addKeyValue(SERVICE_NAME_LOG_KEY, serviceName)
+                            .log("Could not locate Greengrass service to remove service");
+                }
                 kernel.getContext().remove(serviceName);
```

Comment:
this should return the item that it removed, right? We can just use that instead of going through locate

Reply:
remove was returning boolean, updated remove to return the value to avoid calling locate